src/analyses/goto_check_c.cpp

```diff
@@ -2020,6 +2047,10 @@ optionalt<exprt> goto_check_ct::rw_ok_check(exprt expr)
     exprt c = conjunction(conjuncts);
     if(enable_simplify)
       simplify(c, ns);
+
+    // this is a logic expression, C language checks no longer apply
```

Comment:
Possible worth rewording the comment here to be super explicit?
```suggestion
    // this is an internal logic expression to support contracts, not a C language expression.
    // C language checks should not be applied to it
```